database/db.py

```diff
@@ -492,7 +492,8 @@ def remove_friend(self, username: str, friends_username: str) -> None:
         friends_id = friends_id[0][0]
         user_id = user_id[0][0]
         was_successful = self.db_delete(self.query_builder.remove_friend(user_id, friends_id))
-        self.db_delete(self.query_builder.remove_friend(friends_id, user_id))
+        if was_successful:
+            was_successful = all([was_successful, bool(self.db_delete(self.query_builder.remove_friend(friends_id, user_id)))])
```

Comment:
This makes sure that we only report success if both delete operations are successful.